instagram-app/src/components/authentication/withAuthenticate.js

```diff
@@ -0,0 +1,31 @@
+import React from "react";
+
+const withAuthenticate = PostsPage => Login =>
+  class extends React.Component {
+    constructor() {
+      super();
+      this.state = {
+        loggedIn: false
+      };
+    }
+
+    componentDidMount() {
+      if (JSON.parse(localStorage.getItem("user"))) {
```

Comment:
```suggestion
      if (localStorage.getItem("user")) {
```
You don't need JSON.parse here, you only need it if you are trying to keep an entire object on localStorage. Since you are only keeping the User string on localStorage you don't need to use .parse. Also you have user here lowercased when Login was setting User as uppercase, so originally your state wasn't actually being set properly here. I think you should change all instances of User to user anyway.